applications/frontier/src/frontier_ensure_rates_in_syscfg.erl

```diff
@@ -0,0 +1,59 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2010-2013, 2600Hz
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   SIPLABS, LLC (Maksim Krzhemenevskiy)
+%%%-------------------------------------------------------------------
+-module(frontier_ensure_rates_in_syscfg).
+
+-include("frontier.hrl").
+
+%% API
+-export([init/1
+         ,start_link/0
+        ]).
+
+-define(ACCOUNT_RATES_SEC, [{<<"registrations">>, 20}
+                            ,{<<"invites">>, 50}
+                            ,{<<"total_packets">>, 100}
+                           ]).
+-define(DEVICE_RATES_SEC, [{<<"registrations">>, 2}
+                           ,{<<"invites">>, 5}
+                           ,{<<"total_packets">>, 20}
+                           ]).
+-define(ACCOUNT_RATES_MIN, [{<<"registrations">>, 100}
+                            ,{<<"invites">>, 200}
+                            ,{<<"total_packets">>, 2000}
+                           ]).
+-define(DEVICE_RATES_MIN, [{<<"registrations">>, 20}
+                           ,{<<"invites">>, 100}
+                           ,{<<"total_packets">>, 1000}
+                           ]).
+
+-spec sysconfig_default_rates() -> wh_json:object().
+sysconfig_default_rates() ->
+    AccountSec = wh_json:from_list(?ACCOUNT_RATES_SEC),
+    AccountMin = wh_json:from_list(?ACCOUNT_RATES_MIN),
+    AccountRates = wh_json:from_list([{?MINUTE, AccountMin}
+                                      ,{?SECOND, AccountSec}
+                                     ]),
+
+    DeviceSec = wh_json:from_list(?DEVICE_RATES_SEC),
+    DeviceMin = wh_json:from_list(?DEVICE_RATES_MIN),
+    DeviceRates = wh_json:from_list([{?MINUTE, DeviceMin}
+                                      ,{?SECOND, DeviceSec}
+                                     ]),
+
+    wh_json:from_list([{<<"account">>, AccountRates}
+                       ,{<<"device">>, DeviceRates}
+                      ]).
+
+-spec start_link() -> startlink_ret().
+start_link() ->
+    gen_server:start_link({'local', ?MODULE}, ?MODULE, [], []).
```

Comment:
I think start_link() can return 'ignore' instead of calling `gen_server:start_link/4`.

Typically we have a `app_init.erl` (like acdc_init.erl) that will spawn a process to do app initialization stuff (and returning 'ignore' to the supervisor). You may want a frontier_init.erl instead, and have it call to here to init system_config. Then you can also add frontier_maintenance functions to set/update these settings.